core/tee/tee_svc_cryp.c

```diff
@@ -2613,8 +2613,10 @@ TEE_Result syscall_cryp_state_alloc(unsigned long algo, unsigned long mode,
 
 	switch (TEE_ALG_GET_CLASS(algo)) {
 	case TEE_OPERATION_CIPHER:
-		if ((algo == TEE_ALG_AES_XTS && (key1 == 0 || key2 == 0)) ||
-		    (algo != TEE_ALG_AES_XTS && (key1 == 0 || key2 != 0))) {
+		if (((algo == TEE_ALG_AES_XTS || algo == TEE_ALG_SM4_XTS) &&
+		    (key1 == 0 || key2 == 0)) ||
+		    ((algo != TEE_ALG_AES_XTS && algo != TEE_ALG_SM4_XTS) &&
+		    (key1 == 0 || key2 != 0))) {
 			res = TEE_ERROR_BAD_PARAMETERS;
 		} else {
 			res = crypto_cipher_alloc_ctx(&cs->ctx, algo);
```

Comment:
It seems that `crypto_cipher_alloc_ctx()` doesn't know `TEE_ALG_SM4_XTS`.

Reply:
The software computing support for sm4-xts has been added to the second commit.

Reply:
Good, please change the order of the two commits.

Reply:
Is it possible not to change the order of commit, because the first commit is to introduce sm4-xts in the gp process, and the second commit is to implement sm4-xts software computing.

Reply:
Each commit in the git history should build, so you must change the order. You can add the TEE_ALG_SM4_XTS define first in a separate commit.

Reply:
Can I merge these two commit?

Reply:
I would prefer to take them as separate commits.

Reply:
@jenswi-linaro commit "introduce TEE_ALG_SM4_XTS" comes first so both commits build fine. Can you clarify what you expect? Would you like the first commit to introduce the define and touch libutee only, then the second one to add the implementation in core?

Reply:
My bad, the order is good.